src/main/java/org/opensearch/knn/plugin/transport/KNNProfileResponse.java

```diff
@@ -0,0 +1,260 @@
+ *           }
+ *           // ... additional dimensions
+ *         ]
+ *       }
+ *     }
+ *     // ... additional shards
+ *   },
+ *   "cluster_aggregation": {
+ *     "total_shards": 2,
+ *     "dimension": 128,
+ *     "dimensions": [
+ *       {
+ *         "dimension_id": 0,
+ *         "count": 6000,
+ *         "min": -0.723,
+ *         "max": 0.892,
+ *         "mean": 0.172,
+ *         "std_deviation": 0.315,
+ *         "sum": 1032.45,
+ *         "variance": 0.099
+ *       }
+ *       // ... additional dimensions
+ *     ]
+ *   },
+ *   "failures": []
+ * }
+ */
+public class KNNProfileResponse extends BroadcastResponse implements ToXContentObject {
+
+    List<KNNIndexShardProfileResult> shardProfileResults;
+
+    public KNNProfileResponse() {}
+
+    public KNNProfileResponse(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    public KNNProfileResponse(
+        List<KNNIndexShardProfileResult> shardProfileResults,
+        int totalShards,
+        int successfulShards,
+        int failedShards,
+        List<DefaultShardOperationFailedException> shardFailures
+    ) {
+        super(totalShards, successfulShards, failedShards, shardFailures);
+
+        this.shardProfileResults = shardProfileResults;
+    }
+
+    @Override
+    public void writeTo(StreamOutput streamOutput) throws IOException {
+        throw new UnsupportedOperationException("This method is not available");
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+
+        builder.field("total_shards", getTotalShards())
+            .field("successful_shards", getSuccessfulShards())
+            .field("failed_shards", getFailedShards())
+            .startObject("shard_profiles");
+
+        // Add shard profile results
+        for (KNNIndexShardProfileResult shardProfileResult : shardProfileResults) {
+            builder.startObject(shardProfileResult.shardId);
+
+            // Individual segment statistics
+            builder.startArray("segments");
+            for (SegmentProfilerState state : shardProfileResult.segmentProfilerStateList) {
+                builder.startObject()
+                    .field("segment_id", state.getSegmentId())
+                    .field("dimension", state.getDimension())
+                    .startArray("vector_statistics");
+
+                for (int i = 0; i < state.getStatistics().size(); i++) {
+                    SummaryStatistics stats = state.getStatistics().get(i);
+                    builder.startObject()
+                        .field("dimension_index", i)
+                        .startObject("statistics")
+                        .field("count", stats.getN())
+                        .field("min", stats.getMin())
+                        .field("max", stats.getMax())
+                        .field("sum", stats.getSum())
+                        .field("mean", stats.getMean())
```

Comment:
Lets break the logic in this method into separate methods. It will be easier to modifu in future.